db/models/reviews.models.js

```diff
@@ -65,5 +65,25 @@ exports.fetchCommentsByReviewId = (id) => {
       `SELECT * FROM comments WHERE review_id = $1 ORDER BY created_at DESC;`,
       [id]
     )
-    .then(({ rows: comments }) => comments);
+    .then(({ rows: comments }) => {
+      if (!comments)
+        return Promise.reject({
+          status: 404,
+          msg: `No comment found for review_id: ${id}`,
+        });
+      return comments;
+    });
+};
+
+exports.addComment = (id, username, body) => {
+  if (!username || !body)
```

Comment:
The author column in the comments table is a foreign key, so if the provided username didn't correspond to a valid primary key in the users table, PSQL would throw the same error it would do for the valid but not found article_id.

Reply:
Also - your If Statement should have a code block!